### PR TITLE
Release 4.0.0

### LINKED ISSUE #253: [BUG] - MIUI (Xiaomi) cant pick image from gallery
Phone Model: Xiomi Redmi K20 Pro
Version: Android 11 (RQ1.200826.002)
CanHub Version: 3.3.5

I tried your sample project & run it. But it didn't work on my phone.
Also, I uploaded a video for the issue.

https://www.youtube.com/watch?v=MY-bo92LFeo

Please check & replay.

### LINKED ISSUE #274: [BUG] - FileUriExposedException when trying to lunch only camera intent.
- Lib Version 3.3.6
 
**Describe the bug**
App gets crashed when I want to open only camera intent .

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.croppersample/com.canhub.cropper.CropImageActivity}: android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/com.example.croppersample/cache/pickImageResult.jpeg exposed beyond app through ClipData.Item.getUri()

**To Reproduce**
in a crop image sample  use this part of code. 
private val customCropImage = registerForActivityResult(CropImageContract()) {
        ... 
    }

customCropImage.launch(
            options {
                setImagePickerContractOptions(PickImageContractOptions(false,true))
             })

**Media**
Screen Shot 2021-11-23 at 18 30 17

**Smartphone (please complete the following information):**
 - Device: Xiaomi2 mi a2
 - OS: API Level 30 
 
  - Device: Android emulator
 - OS: API Level 28, 29 

@Canato

### PR DESCRIPTION
**Pull request by Canato**
close #271 
close #253
close #274

## Todo List
* [x] CHANGELOG
* [x] README
* [x] Version Number
* [x] Update min version following Android deprecated versions
* [x] Remove all permissions needs for getting pictures, using `Activity Contract`
* [x] Remove previous deprecated methods
* [x] Remove pick image contract
* [x] Improve image source selection layout

# CHANGELOG
### Fixed
- Issue where some devices based on MIUI would not retrieve image from gallery [#253](https://github.com/CanHub/Android-Image-Cropper/issues/253)

### Changed
- `minSdkVersion = 16`, previous versions have been deprecated by Google.

### Removed
- Pick Image, now the library will handle this without expose the contract [#271](https://github.com/CanHub/Android-Image-Cropper/issues/271)
- Need for `READ_EXTERNAL_STORAGE` permission [#271](https://github.com/CanHub/Android-Image-Cropper/issues/271)
- Previous deprecated `ActivityBuilder`, `startPickImageActivity`, `getActivityResult` and `onActivityResult` [#145](https://github.com/CanHub/Android-Image-Cropper/issues/145)

### CODE REVIEW

**Comment by aliaksandr-chaulytka** [comment]
=)
I am waiting for this update 👍

**Comment by Canato** [comment]
> =) I am waiting for this update 👍 CC: @aliaksandr-chaulytka 

Working on this now, close to finish a stable version. Will need a lot of tests hope can count with everyone ^^

**Comment by Canato** [comment]
This version should be working, but I'm not happy with the `AlertDialog` design. 

Planning to change this into a `BottomSheet` format.

File: README.md [hunk 1]
```diff
@@ -162,6 +161,15 @@ override fun onCreate(savedInstanceState: Bundle?) {
...
 
+### Custom dialog for image source pick
+When calling crop directly the library will prompt a dialog for the user choose between gallery or camera (If you keep both enable).
+We use the Android default AlertDialog for this. If you wanna customised it with your app theme you need to override the method `showImageSourceDialog(..)` when extending the activity _(above)_
+```kotlin 
+override fun showImageSourceDialog(openCamera: () -> Unit, openGallery: () -> Unit) {
+     super.showImageSourceDialog(openCamera, openGallery)
+}
+```
+
```
Comment (Canato) [inline README.md:172]: Not final version

File: cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt [hunk 2]
```diff
@@ -98,6 +86,35 @@ open class CropImageActivity :
...
+     * and customised with the app layout if need.
+     */
+    open fun showImageSourceDialog(openCamera: () -> Unit, openGallery: () -> Unit) {
+        // TODO improve layout
+        AlertDialog.Builder(this)
+            .setTitle(R.string.pick_image_chooser_title)
+            .setNegativeButton(R.string.pick_image_camera) { _, _ -> openCamera() }
+            .setPositiveButton(R.string.pick_image_gallery) { _, _ -> openGallery() }
+            .show()
+    }
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt:116]: Improve this layout

**Comment by aliaksandr-chaulytka** [comment]
Really great and fast work, #274 fixed!! Thanks 👍

**Pull request merged**